File: Behavioral/State/Python/yellow_light.py
# yellow_light.py

from traffic_light_state import TrafficLightState
import red_light

class YellowLight(TrafficLightState):
  def handle(self, traffic_light):
    print("Yellow Light - PREPARE!")
    traffic_light.set_state(red_light.RedLight())  # Transition to Red

File: Behavioral/State/JavaScript/yellowLight.js
// yellowLight.js

const TrafficLightState = require("./trafficLightState");

class YellowLight extends TrafficLightState {
  handle(trafficLight) {
    console.log("Yellow Light - PREPARE!");
    
    // FIX: Require inside the method to avoid circular dependency
    const RedLight = require("./redLight");
    trafficLight.setState(new RedLight()); // Transition to Red
  }
}

module.exports = YellowLight;
